src/lib/swapper/swappers/ThorchainSwapper/utils/longTailHelpers.ts

```diff
@@ -53,3 +58,117 @@ export function getTradeType(
+).reduce(
+  (acc, [key, value]) => ({ ...acc, [value]: key }),
+  {} as Record<FeeAmount, AggregatorContract>,
+)
+
+export const generateV3PoolAddressesAcrossFeeRange = (
+  factoryAddress: string,
+  tokenA: Token,
+  tokenB: Token,
+): Map<Address, FeeAmount> => {
+  const poolAddresses = new Map<Address, FeeAmount>()
+  Object.values(FeeAmount)
+    .filter((value): value is FeeAmount => typeof value === 'number')
+    .forEach((fee: FeeAmount) => {
+      const poolAddress = computePoolAddress({
+        factoryAddress,
+        tokenA,
+        tokenB,
+        fee,
+      })
+      poolAddresses.set(poolAddress as Address, fee as FeeAmount)
+    })
+  return poolAddresses
+}
+
+type ContractData = {
+  fee: FeeAmount
+  tokenIn: Address
+  tokenOut: Address
+}
+
+export const getContractDataByPool = async (
+  poolAddresses: Map<Address, FeeAmount>,
+  publicClient: PublicClient,
+  tokenAAddress: string,
+  tokenBAddress: string,
+): Promise<Map<Address, ContractData>> => {
+  const poolContracts = new Map<Address, ContractData>()
+  await Promise.all(
+    Array.from(poolAddresses.entries()).map(async ([address, feeAmount]) => {
+      const poolContract = getContract({
+        abi: IUniswapV3PoolABI,
+        address,
+        publicClient,
+      })
+      try {
+        const [token0, token1]: [Address, Address] = await Promise.all([
+          poolContract.read.token0(),
+          poolContract.read.token1(),
+        ])
+
+        const tokenIn = token0 === tokenAAddress ? token0 : token1
+        const tokenOut = token1 === tokenBAddress ? token1 : token0
+
+        poolContracts.set(poolContract.address, { fee: feeAmount, tokenIn, tokenOut })
+      } catch {
+        // The pool contract is not supported, that's ok - skip it without logging an error
+        return
+      }
+    }),
+  )
+  return poolContracts
+}
+
+export const getQuotedAmountOutByPool = (
+  poolContracts: Map<Address, ContractData>,
+  sellAmount: bigint,
+  quoterContract: GetContractReturnType<typeof QuoterAbi, PublicClient, WalletClient>,
+): Promise<Map<Address, bigint>> => {
+  return Promise.all(
+    Array.from(poolContracts.entries()).map(async ([poolContract, data]) => {
+      const { fee, tokenIn, tokenOut } = data
+      const quotedAmountOut = await quoterContract.simulate
+        .quoteExactInputSingle([tokenIn, tokenOut, fee, sellAmount, BigInt(0)])
+        .then(res => res.result)
+
+      return [poolContract, quotedAmountOut] as [Address, bigint]
+    }),
+  ).then(results => new Map(results))
+}
+
+export const selectBestRate = (
+  quotedAmounts: Map<Address, bigint>,
+): [Address, bigint] | undefined => {
+  return Array.from(quotedAmounts.entries()).reduce(
+    (
+      addressWithHighestAmount: [Address, bigint] | undefined,
+      [poolAddress, amount]: [Address, bigint],
+    ) => {
+      return amount > (addressWithHighestAmount?.[1] ?? BigInt(0))
+        ? [poolAddress, amount]
+        : addressWithHighestAmount
```

Comment:
```suggestion
      if (addressWithHighestAmount === undefined) return [poolAddress, amount]
     
      return amount > addressWithHighestAmount.[1]
        ? [poolAddress, amount]
        : addressWithHighestAmount
```

Reply:
non-blocking of course as they're equivalent it seems

Reply:
Addressed in https://github.com/shapeshift/web/pull/5870/commits/3214c8f434fa895aaa73389850d83f41245609b4